### ISSUE TITLE
Showing hostname in log messages?

### ISSUE THREAD

**Issue by julien51** [elastic/elasticsearch-js#226]
We're trying to debug seemingly "random" errors that we get inside our ElasticSearch Client and I have to say that The JS library makes our job quite difficult:

```javascript
Elasticsearch ERROR: 2015-06-13T14:16:53Z
  Error: Request error, retrying -- socket hang up
      at Log.error (/home/node/trends/node_modules/elasticsearch/src/lib/log.js:213:60)
      at checkRespForFailure (/home/node/trends/node_modules/elasticsearch/src/lib/transport.js:192:18)
      at HttpConnector.<anonymous> (/home/node/trends/node_modules/elasticsearch/src/lib/connectors/http.js:153:7)
      at ClientRequest.wrapper (/home/node/trends/node_modules/elasticsearch/node_modules/lodash/index.js:3128:19)
      at ClientRequest.emit (events.js:95:17)
      at CleartextStream.socketCloseListener (http.js:1522:9)
      at CleartextStream.emit (events.js:117:20)
      at tls.js:691:10
      at process._tickCallback (node.js:419:13)
```

`Request error, retrying -- socket hang up` is quite a limited error message :(

It would be really convenient to have more details :) Things like the hostname (when talking to a cluster, that may help look into the log of the right server...), but also, ideally the last request?... etc

**Comment by spalger** [comment]
I agree, we should do more here

**Comment by saurabhjaluka** [comment]
Any update on this issue? as I am facing the similar problem.

**Comment by spalger** [comment]
Nothing to report now. Will try to take a look soon but this week is pretty packed.

If either of you are interested in giving or a shot the errors are located in [src/lib/errors.js](https://github.com/elastic/elasticsearch-js/blob/master/src/lib/errors.js).

**Comment by saurabhjaluka** [comment]
Okay, thanks. I will try to look into it.

**Comment by jmm** [comment]
@spalger I took a crack at this in #330. `src/lib/errors.js` didn't appear to play a role in this scenario from what I could tell.

**Comment by spalger** [comment]
Closed by #330

**Comment by sahas-** [comment]
Sorry, to comment on a closed issue. I'm seeing the same error with the latest stable release & also on v13.0.0-beta2.. can we re-open this please?

Elasticsearch ERROR: 2017-03-08T03:24:46Z
  Error: Request error, retrying
  POST http://xxxxx/_search => socket hang up
      at Log.error (/Users/sahaswaranamam/../node_modules/elasticsearch/src/lib/log.js:225:56)
      at checkRespForFailure (/Users/sahaswaranamam/../node_modules/elasticsearch/src/lib/transport.js:246:18)
      at HttpConnector.<anonymous> (/Users/sahaswaranamam/../node_modules/elasticsearch/src/lib/connectors/http.js:155:7)
      at ClientRequest.bound (/Users/sahaswaranamam/../node_modules/elasticsearch/node_modules/lodash/dist/lodash.js:729:21)
      at emitOne (events.js:78:13)
      at ClientRequest.emit (events.js:170:7)
      at Socket.socketOnEnd (_http_client.js:295:9)
      at emitNone (events.js:73:20)
      at Socket.emit (events.js:167:7)
      at endReadableNT (_stream_readable.js:906:12)
